client/state/sites/reducer.js

```diff
@@ -125,24 +127,60 @@ export function items( state = {}, action ) {
 			};
 		}
 
-		case SITE_ICON_SET: {
-			const { siteId, iconUrl } = action;
+		case SITE_SETTINGS_UPDATE:
+		case SITE_SETTINGS_RECEIVE: {
```

Comment:
@youknowriad there was a distinction made a long time ago between `site` and `siteSettings` on the API-side wherein `site` requests return quick data but `settings` request get data that is much slower to fetch on the server side. it's unlikely right now that this will change.

> leave the DataLayer optimize this later

I agree with this and think it's a great place to introduce the HTTP request batching, but there are steps involved there, so maybe some clever logic in the API layer can get us there in the meantime